mvpa2/tests/test_mapper.py

```diff
@@ -470,3 +470,41 @@ def test_identity_mapper(s):
     assert_true(idm.forward1(s) is s)
     assert_true(idm.reverse(s) is s)
     assert_true(idm.reverse1(s) is s)
+
+
+def test_static_reverse_doesnt_work_after_feature_selection_tuneup_1():
+    ds_orig = datasets['uni2small'].copy()  # doesn't matter which
+
+    ds = ds_orig.get_mapped(StaticFeatureSelection(np.arange(4)))
+    ds0_rev = ds.a.mapper.reverse1(ds.samples[0])  # should work
+    assert_equal(ds0_rev.shape, (1, ds_orig.nfeatures))
+
+    # direct feature selection
+    ds_ = ds[:, [0, 2]]
+    # should work but doesn't due to
+    # RuntimeError: Cannot reverse-map data since the original data shape is unknown. Either set `dshape` in the constructor, or call train().
+    ds0_rev_ = ds_.a.mapper.reverse1(ds_.samples[0])
+
```

Comment:
aha -- tried to cheat me by my not so obscure use of "_"? ;)  last line should have been
```
assert_equal(ds0_rev_.shape, (ds_orig.nfeatures,))
```
and still not sure if magical autotraining should be done for the StaticFeatureSelection in this case